src/test/java/com/uid2/admin/store/writer/EncryptedScopedStoreWriterTest.java

```diff
@@ -0,0 +1,385 @@
+import static org.mockito.Mockito.*;
+
+
+class EncryptedScopedStoreWriterTest {
+
+    private EncryptedScopedStoreWriter encryptedScopedStoreWriter;
+    private S3Key encryptionKey;
+    private ObjectWriter jsonWriter;
+    private RotatingSiteStore globalStore;
+    private InMemoryStorageMock cloudStorage;
+    private FileManager fileManager;
+    private final List<Site> oneSite = ImmutableList.of(new Site(1, "site 1", true));
+    private final List<Site> anotherSite = ImmutableList.of(new Site(2, "site 2", true));
+    private final String sitesDir = "sites";
+    private final String metadataFileName = "test-metadata.json";
+    private final CloudPath globalMetadataPath = new CloudPath(sitesDir).resolve(metadataFileName);
+    private final GlobalScope globalScope = new GlobalScope(globalMetadataPath);
+    private final FileName dataFile = new FileName("sites", ".json");
+    private final String dataType = "sites";
+    private final int testSiteId = 123;
+
+    @Mock
+    private DownloadCloudStorage fileStreamProvider;
+
+    @Mock
+    private StoreScope scope;
+
+    @Mock
+    private ScopedStoreReader<Map<Integer, S3Key>> reader;
+
+    @Mock
+    private Clock clock;
+
+    @Mock
+    private VersionGenerator versionGenerator;
+
+    @Mock
+    private IMetadataVersionedStore provider;
+
+    @Mock
+    private RotatingS3KeyProvider s3KeyProvider;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        cloudStorage = new InMemoryStorageMock();
+        FileStorageMock fileStorage = new FileStorageMock(cloudStorage);
+        fileManager = new FileManager(cloudStorage, fileStorage);
+        globalStore = mock(RotatingSiteStore.class);
+        jsonWriter = JsonUtil.createJsonWriter();
+
+        // Generate a valid 32-byte AES key
+        byte[] keyBytes = Random.getRandomKeyBytes();
+        String base64Key = Base64.getEncoder().encodeToString(keyBytes);
+        encryptionKey = new S3Key(1, testSiteId, 0, 0, base64Key);
+
+        Map<Integer, S3Key> mockS3Keys = new HashMap<>();
+        mockS3Keys.put(testSiteId, encryptionKey);
+
+        // Mock the reader
+        when(reader.getSnapshot()).thenReturn(mockS3Keys);
+
+        // Mock the s3KeyProvider
+        when(s3KeyProvider.getAll()).thenReturn(mockS3Keys);
+
+        // Mock the globalStore
+        when(globalStore.getAllSites()).thenReturn(new ArrayList<>());
+        when(globalStore.getMetadata()).thenReturn(new JsonObject());
+
+        // Initialize EncryptedScopedStoreWriter with the s3KeyProvider
+        encryptedScopedStoreWriter = new EncryptedScopedStoreWriter(
+                provider,
+                fileManager,
+                versionGenerator,
+                clock,
+                scope,
+                dataFile,
+                dataType,
+                s3KeyProvider
+        );
+    }
+
+    @Test
+    void testEncryptData() throws Exception {
```

Comment:
This test doesn't actually call any methods on encryptedScopedStoreWriter - which is the target of the tests. 
Duplicating the logic and then testing that doesn't help us test that the class does what is expected.